src/main/java/seedu/address/logic/commands/DeleteWeddingCommand.java

```diff
@@ -45,11 +62,38 @@ public CommandResult execute(Model model) throws CommandException {
                     return new CommandResult(String.format(MESSAGE_DELETE_WEDDING_SUCCESS,
                             Messages.format(targetWedding)));
                 } else {
-                    throw new CommandException(
-                            String.format(MESSAGE_DELETE_WEDDING_FAILURE_STILL_USED, Messages.format(targetWedding)));
+                    if (this.force) {
+                        for (Person person : model.getFilteredPersonList()) {
+                            HashSet<Wedding> personWeddings = new HashSet<>(person.getWeddings());
+                            if (personWeddings.contains(wedding)) {
+                                personWeddings.remove(wedding);
+                                Person newPerson = new Person(
+                                        person.getName(),
+                                        person.getPhone(),
+                                        person.getEmail(),
+                                        person.getAddress(),
+                                        person.getTags(),
+                                        personWeddings
+                                );
+                                model.setPerson(person, newPerson);
+                                model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+                            }
+                        }
+                        model.deleteWedding(wedding);
+                        return new CommandResult(String.format(MESSAGE_DELETE_WEDDING_SUCCESS,
+                                Messages.format(targetWedding)));
+                    } else {
+                        throw new CommandException(
+                                String.format(MESSAGE_DELETE_WEDDING_FAILURE_STILL_USED,
+                                        Messages.format(targetWedding))
+                                + ".\n"
+                                + Messages.MESSAGE_FORCE_DELETE_WEDDING
+                        );
+                    }
                 }
             }
         }
+        model.updateFilteredWeddingList(Model.PREDICATE_SHOW_ALL_WEDDINGS);
         throw new CommandException(String.format(MESSAGE_DELETE_WEDDING_FAILURE_NOT_FOUND,
```

Comment:
I think should add a check that `this.force` equals the others' `force` for `DeleteWeddingCommand` and `DeleteTagCommand` in the `equals` method - same for `toString()`, can add force to the StringBuilder

Reply:
Fixed!